agent/api/task/service_connect.go

```diff
@@ -65,3 +80,55 @@ type DNSConfigEntry struct {
 	HostName string `json:"hostName"`
 	Address  string `json:"address"`
 }
+
+// ExtractSCConfigFromACSAttachment finds the service connect attachment, and extracts the service connect container name and service connect config value.
+func ExtractSCConfigFromACSAttachment(acsTask *ecsacs.Task) (*ServiceConnectConfig, error) {
+	if acsTask.Attachments == nil {
+		return nil, nil
+	}
+	var scAttachment *ecsacs.Attachment
+	containerName := ""
+	for _, attachment := range acsTask.Attachments {
+		if aws.StringValue(attachment.AttachmentType) == serviceConnectAttType {
+			scAttachment = attachment
+			break
+		}
+	}
+	// no service connect attachment is found
+	if scAttachment == nil {
+		return nil, nil
+	}
+	// translate ecsacs.Attachment.Value to ServiceConnectConfig
+	scConfigValue := &ServiceConnectConfig{}
+	for _, property := range scAttachment.AttachmentProperties {
+		switch aws.StringValue(property.Name) {
+		case serviceConnectAttProSCConfigName:
+			s, _ := strconv.Unquote(string(*property.Value))
+			if err := json.Unmarshal([]byte(s), scConfigValue); err != nil {
+				return nil, err
+			}
+		case serviceConnectAttProSCContainerName:
+			containerName = aws.StringValue(property.Value)
+		}
+	}
+	scConfigValue.ContainerName = containerName
+	if err := ValidateSCConfig(scConfigValue); err != nil {
+		return nil, err
+	}
+	return scConfigValue, nil
+}
+
+func ValidateSCConfig(scConfig *ServiceConnectConfig) error {
+	if scConfig.ContainerName == "" {
+		return errors.Errorf("service connect config validation: no service connect ContainerName")
+	}
+
+	if len(scConfig.IngressConfig) == 0 {
+		return errors.Errorf("service connect config validation: no service connect IngressConfig")
+	}
+
+	if len(scConfig.DNSConfig) == 0 {
```

Comment:
I believe this is allowed to be empty. It would be weird, but I seems ok to have a container that won't do outbound SC calls, but only receive inbound ones.

Also, I think we need to validate more in depth here (this goes for all other options), for example, are IPs valid, port numbers, etc.

Reply:
+1. We also need to validate Egress config

Reply:
Validations are added to commit 3d8d95406e74fc25d3c00d384a11cd5cd4ae55f9. Please take a look on `service_connect_validation.go`. Thanks.